fbgemm_gpu/src/dram_kv_embedding_cache/dram_kv_embedding_cache.h

```diff
@@ -249,28 +327,33 @@ class DramKVEmbeddingCache : public kv_db::EmbeddingKVDB {
                             block = it->second;
                           } else {
                             // Key doesn't exist, allocate new block and insert.
-                            block = StoreValueUtils::allocate<weight_type>(
-                                block_size_, block_alignment_, pool);
+                            block = pool->allocate_t();
                             wlmap->insert({id, block});
                           }
-                          StoreValueUtils::update_timestamp<weight_type>(
-                              block, now);
+                          if (feature_evict_config_.trigger_mode !=
+                                  EvictTriggerMode::DISABLED &&
+                              feature_evict_) {
+                            feature_evict_->update_feature_statistics(block);
+                          }
                           auto* data_ptr =
-                              StoreValueUtils::data_ptr<weight_type>(block);
-                          std::copy(
-                              weights[id_index]
-                                  .template data_ptr<weight_type>(),
-                              weights[id_index]
-                                      .template data_ptr<weight_type>() +
-                                  weights[id_index].numel(),
-                              data_ptr);
+                              FixedBlockPool::data_ptr<weight_type>(block);
+                          std::copy(weights[id_index]
+                                        .template data_ptr<weight_type>(),
+                                    weights[id_index]
+                                            .template data_ptr<weight_type>() +
+                                        weights[id_index].numel(),
```

Comment:
from here, seems the weight id is not copied to block, where will  you copy weight id to the block?